### PR TITLE
Update Readme and go build

### PR DESCRIPTION
**Pull request by saikat-royc**
1. Update README matrix
2. use mod=vendor for go build

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,15 +22,15 @@ Latest image: `gke.gcr.io/gcp-filestore-csi-driver:v0.2.0`
...
 ### Kubernetes Compatibility
 
-| Filestore CSI Driver\Kubernetes Version | 1.10.5-1.11 | 1.12 | 1.13 | 1.14 |
-| --------------------------------------- | ----------- | ---- | ---- | ---- |
-| v0.1.0 (alpha)                          | yes         | no   | no   | no   |
-| v0.2.0 (alpha)                          | no          | yes  | yes  | yes  |
-| dev                                     | no          | yes  | yes  | yes  |
+| Filestore CSI Driver\Kubernetes Version | 1.10.5-1.11 | 1.12 | 1.13 | 1.14 | 1.15 | 1.16+ |
+| --------------------------------------- | ----------- | ---- | ---- | ---- | ---- | ----  |
+| v0.1.0 (alpha)                          | yes         | no   | no   | no   |  no  |  no   |
```
Comment (saikat-royc) [inline README.md:31]: I am not sure if these are needed? 0.1.0 and 0.2.0.
@msau42

Comment (msau42) [reply README.md:31]: Yeah I think we can remove v0.1.0 information and eventually v0.2.0 when we make a new release.

Kubenretes 1.14 and lower is already out of support from the community

Comment (saikat-royc) [reply README.md]: Ack

File: README.md [hunk 2]
```diff
@@ -22,15 +22,14 @@ Latest image: `gke.gcr.io/gcp-filestore-csi-driver:v0.2.0`
...
+This plugin is compatible with CSI version 1.3.0.
 
 ### Kubernetes Compatibility
 
-| Filestore CSI Driver\Kubernetes Version | 1.10.5-1.11 | 1.12 | 1.13 | 1.14 |
-| --------------------------------------- | ----------- | ---- | ---- | ---- |
-| v0.1.0 (alpha)                          | yes         | no   | no   | no   |
-| v0.2.0 (alpha)                          | no          | yes  | yes  | yes  |
-| dev                                     | no          | yes  | yes  | yes  |
+| Filestore CSI Driver\Kubernetes Version | 1.10.5-1.11 | 1.12 | 1.13 | 1.14 | 1.15 | 1.16+ |
```
Comment (msau42) [inline README.md:29]: we can remove the 1.10.5-1.11 column and just start from 1.12 for now

Comment (saikat-royc) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (mattcary) [reply README.md]: +1

**Pull request merged**